mautic/mautic
# Fix PHP Notice on show_php_version_warning_if_under

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | 3.2
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | N/A
| Related developer documentation PR URL | N/A
| Issue(s) addressed                     | Fixes #9619




#### Description:

We added `show_php_version_warning_if_under` in Mautic 3.2.5. However, the `mautic:update:find` action will also fetch previous releases from GitHub which don't have this key in their `app/release_metadata.json`, hence triggering PHP notices. This PR fixes that issue by first checking if `show_php_version_warning_if_under` exists in the metadata.



#### Steps to test this PR:
1. Load up this PR locally (you can't test this on Mautibox)
2. Run `bin/console mautic:update:find`. Before applying this PR, you'll likely see PHP notices:
```
dennis@mautic3-web:/var/www/html$ bin/console mautic:update:find
PHP Notice:  Undefined index: show_php_version_warning_if_under in /var/www/html/app/bundles/CoreBundle/Release/Metadata.php on line 82

Notice: Undefined index: show_php_version_warning_if_under in /var/www/html/app/bundles/CoreBundle/Release/Metadata.php on line 82
PHP Notice:  Undefined index: show_php_version_warning_if_under in /var/www/html/app/bundles/CoreBundle/Release/Metadata.php on line 82

Notice: Undefined index: show_php_version_warning_if_under in /var/www/html/app/bundles/CoreBundle/Release/Metadata.php on line 82
Great! You are running the current version of Mautic.
```

After applying this PR, the PHP Notices should be gone:

```
dennis@mautic3-web:/var/www/html$ bin/console mautic:update:find
Great! You are running the current version of Mautic.
```
